### DISCUSSION TITLE
What is the difference between "citation-key" and "id" in CSL YAML export?

### DISCUSSION THREAD

**Discussion by allefeld** [retorquere/zotero-better-bibtex#2943] [General]
When I export to CSL YAML, the entries always start with a field `id`, and later comes a field `citation-key`. E.g.

```yaml
- id: ernst2004
  …
  citation-key: ernst2004
  …
```

The two keys seem to be usually, but not always, the same.

My questions:

- Why are there two similar (identical?) keys in the YAML output, and what is their difference?
- Which of the two should be used for citation e.g. in Pandoc Markdown?
- Could the one not used for citation be omitted from the export?

**Comment by retorquere** [comment]
> The two keys seem to be usually, but not always, the same.

They ought to be the same, always. 

> My questions:
> 
> - Why are there two similar (identical?) keys in the YAML output, and what is their difference?

Different purposes. citation-key is a CSL variable that can be used in rendered output, in principle (although I don't know any style which does). id is how you address the item when citing in eg pandoc. 

> - Which of the two should be used for citation e.g. in Pandoc Markdown?

id. 

> - Could the one not used for citation be omitted from the export?

Yes, as can any CSL field, by adding it to skipfields in the preferences.

**Reply by allefeld** [reply]
Thanks!
